modules/test/playwright/fixtures/login.fixture.ts

```diff
@@ -0,0 +1,107 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Cookie, Page, expect, test} from '@playwright/test';
+
+import {liferayConfig} from '../liferay.config';
+import {createTempFile, readTempFile} from '../utils/util';
+
+export interface Login {
+	password: string;
+	sessionId: string;
+	user: string;
+}
+
+const TIMEOUT_OPTION = {
+	timeout: 5 * 1000,
+};
+
+let loggedIn = false;
+
+const loginTest = test.extend<{
+	_login: Login;
+}>({
+	_login: async ({page}, use) => {
+		const user = liferayConfig.user.login;
+		const password = liferayConfig.user.password;
+
+		if (!loggedIn) {
+			const storageStatePath = createTempFile('storageState.json');
+
+			await page.goto('/');
+
+			await page.getByRole('button', {name: 'Sign In'}).click();
+
+			await page.getByLabel('Email Address').fill(user);
+			await page.getByLabel('Password').fill(password);
+			await page.getByLabel('Remember Me').check();
+
+			await page
+				.getByLabel('Sign In- Loading')
+				.getByRole('button', {name: 'Sign In'})
+				.click();
+
+			await maybeSetRecoveryAnswer(page);
+
+			await maybeAcceptTermsOfUse(page);
+
+			await expect(
+				page.getByLabel('Open Applications MenuCtrl+')
+			).toBeVisible(TIMEOUT_OPTION);
+
+			await page.context().storageState({path: storageStatePath});
+
+			loggedIn = true;
```

Comment:
What if login fails? Does the entire fixture error out early before setting this to true?